### PR TITLE
.Net: Move static factory methods to SKFunction / SKPlugin

### PR DESCRIPTION
**Pull request by stephentoub**
Moves the recently added CreateFunctionFromXx and CreatePluginFromXx methods to SKFunction and SKPlugin and renames them as discussed.

No implementations changed here; the code just moved around.

### CODE REVIEW

**Comment by stephentoub** [comment]
@markwallace-microsoft, @SergeyMenshykh, please take a look when you have a moment.

File: dotnet/src/Experimental/Orchestration.Flow.UnitTests/FlowExtensionsTests.cs [hunk 1]
```diff
@@ -8,6 +8,8 @@
 using Microsoft.SemanticKernel.Experimental.Orchestration.Abstractions;
 using Xunit;
 
+#pragma warning disable SKEXP0001
```
Comment (SergeyMenshykh) [inline dotnet/src/Experimental/Orchestration.Flow.UnitTests/FlowExtensionsTests.cs:11]: @stephentoub what are this warnings are about? and why are we disabling them?

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/src/Experimental/Orchestration.Flow.UnitTests/FlowExtensionsTests.cs (hunk 1 above)
Comment (stephentoub) [reply dotnet/src/Experimental/Orchestration.Flow.UnitTests/FlowExtensionsTests.cs:11]: The experimental types being used here are marked [Experimental]. This triggers a warning to let folks know they're depending on something that might change. The tests need to suppress that warning.

Comment (SergeyMenshykh) [reply dotnet/src/Experimental/Orchestration.Flow.UnitTests/FlowExtensionsTests.cs:11]: Got it, thanks.

**Pull request merged**